src/map_publisher.cpp

```diff
@@ -52,18 +54,64 @@ using namespace std::chrono_literals;
 class MapPublisher : public rclcpp::Node {
  public:
   MapPublisher() : Node("map_publisher") {
-    std::string file_path = this->declare_parameter("tif_path", ".");
-    std::string color_path = this->declare_parameter("tif_color_path", ".");
-    std::string frame_id = this->declare_parameter("frame_id", "map");
-
     original_map_pub_ = this->create_publisher<grid_map_msgs::msg::GridMap>("elevation_map", 1);
+    const std::string frame_id = this->declare_parameter("frame_id", "map");
+    const std::string gdal_dataset_path = this->declare_parameter("gdal_dataset_path", ".");
+    const std::string color_path = this->declare_parameter("gdal_dataset_color_path", COLOR_MAP_DEFAULT_PATH);
+    rcl_interfaces::msg::ParameterDescriptor max_map_descriptor;
+    max_map_descriptor.read_only = true;
+    max_map_descriptor.description =
+        "Maximum number of raster pixels able to be loaded. \
+        Useful when working with large raster datasets to limit memory usage. \
+        Set to 0 to disable limits (may cause runtime crash).";
+    rcl_interfaces::msg::IntegerRange max_map_descriptor_int_range;
+
+    max_map_descriptor_int_range.from_value = 0;
+    max_map_descriptor_int_range.to_value = std::numeric_limits<int>::max();
+    max_map_descriptor.integer_range.push_back(max_map_descriptor_int_range);
+
+    const uint16_t max_map_width =
+        std::clamp(this->declare_parameter("max_map_width", 1024, max_map_descriptor),
+                   max_map_descriptor_int_range.from_value, max_map_descriptor_int_range.to_value);
+    const uint16_t max_map_height =
+        std::clamp(this->declare_parameter("max_map_height", 1024, max_map_descriptor),
+                   max_map_descriptor_int_range.from_value, max_map_descriptor_int_range.to_value);
 
-    RCLCPP_INFO_STREAM(get_logger(), "file_path " << file_path);
-    RCLCPP_INFO_STREAM(get_logger(), "color_path " << color_path);
+    RCLCPP_INFO_STREAM(get_logger(), "gdal_dataset_path: '" << gdal_dataset_path << "'");
+    RCLCPP_INFO_STREAM(get_logger(), "gdal_dataset_color_path: '" << color_path << "'");
     tf_broadcaster_ = std::make_shared<tf2_ros::StaticTransformBroadcaster>(this);
 
     map_ = std::make_shared<GridMapGeo>(frame_id);
-    map_->Load(file_path, color_path);
+    map_->setMaxMapSizePixels(max_map_width, max_map_height);
```

Comment:
What happens if there are no parameters defined? Shouldn't it just use whatever the tif file has?

Reply:
Ah yea, that sounds like better behavior. I'll add that in.

Reply:
I changed the defaults to be infinite width and height, so it just uses what the raster size is.

For loading large rasters like the SRTM VRT, you must configure limits to not have a runtime memory allocation crash.